xdsl/interactive/app.py

```diff
@@ -249,10 +251,37 @@ def compute_available_pass_list(self) -> tuple[AvailablePass, ...]:
             case Exception():
                 return ()
             case ModuleOp():
+                # transform rewrites into passes
+                rewrites = get_all_possible_rewrites(
+                    self.current_module,
+                    individual_rewrite.REWRITE_BY_NAMES,
+                )
+                rewrites_as_pass_list: tuple[AvailablePass, ...] = ()
+                for op_idx, (op_name, pat_name) in rewrites:
+                    rewrite_pass = individual_rewrite.IndividualRewrite
+                    rewrite_spec_arg_str = f'matched_operation_index={op_idx} operation_name="{op_name}" pattern_name={pat_name}'
+                    rewrite_spec = list(
+                        parse_pipeline(f"{rewrite_pass.name}{{{rewrite_spec_arg_str}}}")
+                    )[0]
+                    op = list(self.current_module.walk())[op_idx]
```

Comment:
I add the op to what "get_all_possible_rewrites" returns. what do u think? does this make sense?

Reply:
I considered removing the operation index, but I think we should leave it for now as it could be useful for future plans we have for functionalities (line numbers etc.)

Reply:
@superlopuh I realize now that maybe it is better to leave it the way that it is. 

returning the `str(Operation)` from get_all_possible_rewrites():

When the rewrite is displayed in the gui, we now get

`%0 = arith.addi %1, %2`
instead of
`Addi(%res = arith.addi %two, %n : i32)`


returning the `Operation` from get_all_possible_rewrites() - (This wont work either way due to issues with testing and equality checking of an Operation):

When the rewrite is displayed in the gui, we now get

`Addi(%0 = arith.addi %1, %2)`
instead of
`Addi(%res = arith.addi %two, %n : i32)`

Reply:
https://github.com/xdslproject/xdsl/issues/2162 
@superlopuh told me to file an issue. we will address this later. will merge as is.